library/src/test/java/com/novoda/downloadmanager/NetworkFileDownloaderTest.java

```diff
@@ -0,0 +1,120 @@
+package com.novoda.downloadmanager;
+
+import com.novoda.notils.logger.simple.Log;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static com.novoda.downloadmanager.InternalFileSizeFixtures.aFileSize;
+import static com.novoda.downloadmanager.NetworkResponseFixtures.aNetworkResponse;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.*;
+
+public class NetworkFileDownloaderTest {
+
+    private static final byte[] BYTES_TO_RECEIVE = "s".getBytes();
+
+    private static final String ANY_RAW_URL = "http://example.com";
+    private static final FileSize KNOWN_FILE_SIZE = aFileSize().withAreBytesDownloadedKnown(true).build();
+    private static final FileSize UNKNOWN_FILE_SIZE = aFileSize().withAreBytesDownloadedKnown(false).build();
+
+    private final FileDownloader.Callback callback = mock(FileDownloader.Callback.class);
+    private final HttpClient httpClient = mock(HttpClient.class);
+    private final NetworkRequestCreator requestCreator = spy(new NetworkRequestCreator());
+
+    private NetworkFileDownloader networkFileDownloader;
+
+    @Before
+    public void setUp() throws IOException {
+        Log.setShowLogs(false);
+        networkFileDownloader = new NetworkFileDownloader(httpClient, requestCreator);
+        given(httpClient.execute(any(NetworkRequest.class))).willReturn(
+                aNetworkResponse().withInputStream(new ByteArrayInputStream(BYTES_TO_RECEIVE)).build()
+        );
+    }
+
+    @Test
+    public void createsRequestWithDownloadedBytesHeader_whenKnown() {
+        networkFileDownloader.startDownloading(ANY_RAW_URL, KNOWN_FILE_SIZE, callback);
+
+        verify(requestCreator).createDownloadRequestWithDownloadedBytesHeader(ANY_RAW_URL, KNOWN_FILE_SIZE.currentSize(), KNOWN_FILE_SIZE.totalSize());
```

Comment:
I think that this is testing an interaction with an internal collaborator. If we refactor this class and the way that the internal collaborators work we will then have to come here and change this again. We should only test what the public methods/callbacks return and interaction with external collaborators. What do you think? The idea is to allow internal refactoring without having to change tests, while keeping behaviour tested

Reply:
👍 very true. Updated in 2703344